crates/forge_analyzer/src/definitions.rs

```diff
@@ -960,6 +960,21 @@ impl<'cx> FunctionAnalyzer<'cx> {
                     ApiCallKind::Authorize => Some(Intrinsic::Authorize(IntrinsicName::Other)),
                 }
             }
+
+            [PropPath::Def(def), PropPath::Static(ref s), ..] if is_storage_read(s) => {
+                match self.res.is_imported_from(def, "@forge/api") {
+                    Some(ImportKind::Named(ref name)) if *name == *"storage" => {
+                        Some(Intrinsic::StorageRead)
+                    }
+                    _ => None,
+                }
+            }
+            [PropPath::Def(def), ..] => match self.res.is_imported_from(def, "@forge/api") {
+                Some(ImportKind::Named(ref name)) if *name == *"authorize" => {
+                    Some(Intrinsic::Authorize(IntrinsicName::Other))
+                }
+                _ => None,
+            },
```

Comment:
Specifically moving this case above the secret scanner checks I think catches all statements before secret scanner checks.

Reply:
Ah, mb I thought that match was a guard statement for some reason. If you change it to a guard statement with `if self.res.is_imported_from(def, "@forge/api") == *"authorize" => Some(Intrinsic::Authorize(IntrinsicName::Other)),`, then moving it above shouldn't swallow the secret scanner checks.